src/lib.rs

```diff
@@ -511,3 +555,315 @@ where
 
     Ok(value)
 }
+
+async fn consume_value_buffers(
+    partition_assignment: Arc<Mutex<PartitionAssignment>>,
+    buffers: &mut ValueBuffers,
+) -> Result<(Vec<Value>, HashMap<DataTypePartition, DataTypeOffset>), KafkaJsonToDeltaError> {
+    let mut partition_assignment = partition_assignment.lock().await;
+
+    let ConsumedBuffers {
+        values,
+        partition_offsets,
+    } = buffers.consume();
```

Comment:
what the hell is this wizardry.

I can't say I've seen this syntax before :exploding_head:

Reply:
This is a destructuring of the struct returned by `consume` - a la https://doc.rust-lang.org/rust-by-example/flow_control/match/destructuring/destructure_structures.html